java/client/src/main/java/glide/api/models/commands/FT/FTProfileOptions.java

```diff
@@ -0,0 +1,175 @@
+    private final GlideString[] query;
+
+    /** Query type being profiled. */
+    public enum QueryType {
+        SEARCH,
+        AGGREGATE
+    }
+
+    /**
+     * Profile a query given as an array of module command line arguments.
+     *
+     * @param queryType The query type.
+     * @param commandLine Command arguments (not including index name).
+     */
+    public FTProfileOptions(@NonNull QueryType queryType, @NonNull GlideString[] commandLine) {
+        this(queryType, commandLine, false);
+    }
+
+    /**
+     * Profile a query given as an array of module command line arguments.
+     *
+     * @param queryType The query type.
+     * @param commandLine Command arguments (not including index name).
+     */
+    public FTProfileOptions(@NonNull QueryType queryType, @NonNull String[] commandLine) {
+        this(queryType, commandLine, false);
+    }
+
+    /**
+     * Profile a query given as an array of module command line arguments.
+     *
+     * @param queryType The query type.
+     * @param commandLine Command arguments (not including index name).
+     * @param limited Either provide a full verbose output or some brief version (limited).
+     */
+    public FTProfileOptions(
+            @NonNull QueryType queryType, @NonNull GlideString[] commandLine, boolean limited) {
+        this.queryType = queryType;
+        this.query = commandLine;
+        this.limited = limited;
+    }
+
+    /**
+     * Profile a query given as an array of module command line arguments.
+     *
+     * @param queryType The query type.
+     * @param commandLine Command arguments (not including index name).
+     * @param limited Either provide a full verbose output or some brief version (limited).
+     */
+    public FTProfileOptions(
+            @NonNull QueryType queryType, @NonNull String[] commandLine, boolean limited) {
+        this(
+                queryType,
+                Stream.of(commandLine).map(GlideString::gs).toArray(GlideString[]::new),
+                limited);
+    }
+
+    /**
+     * Profile an aggregation query with given parameters.
+     *
+     * @param query The query itself.
+     * @param options {@link FT#aggregate} options.
+     */
+    public FTProfileOptions(@NonNull String query, @NonNull FTAggregateOptions options) {
+        this(gs(query), options);
+    }
+
+    /**
+     * Profile an aggregation query with given parameters.
+     *
+     * @param query The query itself.
+     * @param options {@link FT#aggregate} options.
+     */
+    public FTProfileOptions(@NonNull GlideString query, @NonNull FTAggregateOptions options) {
+        this(QueryType.AGGREGATE, concatenateArrays(new GlideString[] {query}, options.toArgs()));
+    }
+
+    /**
+     * Profile a search query with given parameters.
+     *
+     * @param query The query itself.
+     * @param options {@link FT#search} options.
+     */
+    public FTProfileOptions(@NonNull String query, @NonNull FTSearchOptions options) {
```

Comment:
Same as above. Why do we need FTSearchOptions? It is not present in the documentation.

Reply:
The command accepts all arguments of aggregate/search and does the same action + profiling.